### PR TITLE
docs: [IAC-2104]: Add IaCM built-in policies 

### PR DESCRIPTION
**Pull request by richardblack-Harness**
## Description

* Add new IaCM-specific policies section with Terraform Plan Cost example
* [JIRA](https://harness.atlassian.net/browse/IAC-2104)

PRs must meet these requirements to be merged:
- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [x] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/infra-as-code-management/policies/terraform-plan-cost-policy.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    ></iframe>
+</TabItem>
+<TabItem value="Step-by-step">
+To add a new Terraform Plan Cost policy and assign it to a policy set, follow these steps:
+
+
+### Add a new policy
+1. Sign in to [app.harness.io](https://app.harness.io), then select **Project Settings**.
+2. Scroll down and select **Policies**.
+3. Select **Policies**, then select **New Policy**.
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:37]: Do you have to select Policies twice (steps 2 and 3)?

File: docs/infra-as-code-management/policies/terraform-plan-cost-policy.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+7. Inspect the sample and edit the amount accordingly.
+8. Select **Test** to test the policy against the provided sample data, focusing specifically on **TotalMonthlyCost**.
+9. Select **Save**.
+
+### Add the policy to a policy set
+1. Select **Policy sets**.
+2. Select **New policy set**.
+3. Name your policy set.
+4. Select **Terraform Plan Cost** from the entity type list.
+5. Select **Continue**
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:50]:
```suggestion
5. Select **Continue**.
```

File: docs/infra-as-code-management/policies/terraform-plan-cost-policy.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        referrerpolicy="strict-origin-when-cross-origin"
+        frameborder="0"
+        webkitallowfullscreen="true"
+        mozallowfullscreen="true"
+        allowfullscreen="true"
+    ></iframe>
+</TabItem>
+<TabItem value="Step-by-step">
+To add a new Terraform Plan Cost policy and assign it to a policy set, follow these steps:
+
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:32]:
```suggestion
```

File: docs/infra-as-code-management/policies/terraform-plan-cost-policy.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    height="100%"
+    referrerpolicy="strict-origin-when-cross-origin"
+    frameborder="0"
+    webkitallowfullscreen="true"
+    mozallowfullscreen="true"
+    allowfullscreen="true"
+></iframe>
+</TabItem>
+<TabItem value="Step-by-step">
+ To compare your previously execution Terraform plan step against a policy, follow these steps:
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:74]:
```suggestion
 To compare your previously executed Terraform plan step against a policy, follow these steps:
```

File: docs/infra-as-code-management/policies/terraform-plan-cost-policy.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Select **Policy sets** then select **Policies**.
+2. Select **Policies**, then select the policy you want to review your previously executed plan step against.
+3. Click **Select input**.
+4. Select the **Entity type**, **Organization**, **Project** and **Action** to search for filter your pipeline executions.
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:79]:
```suggestion
4. Select the **Entity type**, **Organization**, **Project**, and **Action** to search for and filter your pipeline executions.
```

File: docs/infra-as-code-management/policies/terraform-plan-cost-policy.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+5. Select **Apply**.
+6. Review your policy TotalMonthlyCost amount against the TotalMonthlyCost of your Terraform plan input.
+</TabItem>
+</Tabs>
+
+## Conclusion
+
+To summarize, you can easily add and customize Harness built-in policies and enforce them so that they run against your execution pipelines. Harness provides convenient test data to run your policy against before saving and enforcing it. After your pipeline is executed, you can review its plan data specifically again a policy.
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:87]:
```suggestion
```

Comment (brian-f-harness) [reply docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:87]: We normally don't have a Conclusion to topics. You could add some of this verbiage to the intro instead. If your docs normally include a Conclusion, please disregard this suggestion.

File: docs/infra-as-code-management/policies/_category_.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{
+    "label":"Policies & Governance",
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/_category_.json:2]:
```suggestion
    "label":"Policies and Governance",
```

File: docs/infra-as-code-management/policies/_category_.json [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    "position": 50,
+    "collapsible":"true",
+    "collapsed":"true",
+    "className":"red",
+    "link":{
+       "type":"generated-index",
+       "title":"Policies & Governance",
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/_category_.json:9]:
```suggestion
       "title":"Policies and Governance",
```

File: docs/infra-as-code-management/policies/terraform-plan-cost-policy.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
+---
+title: Built-in policies
+description: A guide to show you how to add a terraform plan cost policy
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Policies can be added and assigned to policy sets to add protective steps that run in conjunction with your IaCM pipelines, evaluating changes and either forcing the pipeline to fail or warning you about conditions that have not been met by your IaCM policy set.
```
Comment (brian-f-harness) [inline docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:9]:
```suggestion
You can add policies and assign them to policy sets to add protective steps that run in conjunction with your IaCM pipelines, evaluating changes and either forcing the pipeline to fail or warning you about conditions that have not been met by your IaCM policy set.
```

**Review by brian-f-harness — APPROVED** [review]
left a few minor suggestions, otherwise LGTM

File: docs/infra-as-code-management/policies/terraform-plan-cost-policy.md (hunk 1 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/policies/terraform-plan-cost-policy.md:37]: I cringed a little at this point, too, but yes, policies is selected twice.

https://github.com/harness/developer-hub/assets/167115273/9e542169-3369-412e-aea9-831c2226de70

File: docs/infra-as-code-management/policies/_category_.json (hunk 7 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/policies/_category_.json:2]: Updated this to Built-in policies

File: docs/infra-as-code-management/policies/_category_.json (hunk 8 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/policies/_category_.json:9]: Updated this to Built-in policies

**Review by brian-f-harness — APPROVED** [review]

**Pull request merged**